src/openrct2-ui/input/MouseInput.cpp

```diff
@@ -109,26 +109,27 @@ void game_handle_input()
 
     invalidate_all_windows_after_input();
 
-    int32_t x, y, state;
-    while ((state = game_get_next_input(&x, &y)) != MOUSE_STATE_RELEASED)
+    int32_t state;
+    ScreenCoordsXY screenCoords;
+    while ((state = game_get_next_input(screenCoords)) != MOUSE_STATE_RELEASED)
     {
-        game_handle_input_mouse(x, y, state & 0xFF);
+        game_handle_input_mouse(screenCoords, state & 0xFF);
     }
 
     if (_inputFlags & INPUT_FLAG_5)
     {
-        game_handle_input_mouse(x, y, state);
+        game_handle_input_mouse(screenCoords, state);
     }
-    else if (x != MONEY32_UNDEFINED)
+    else if (screenCoords.x != MONEY32_UNDEFINED)
```

Comment:
This looks v. odd. We should find out if this information can be passed back in a nicer manor. Might not be appropriate to change it in this refactor though.

Reply:
This looks like dead code, no one ever assigns MONEY32_UNDEFINED to the coordinates.

Reply:
It is is very, if you think it's worth investigating, open an issue and I can follow up on that later.

Reply:
I've created the issue https://github.com/OpenRCT2/OpenRCT2/issues/10121